code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm

```diff
@@ -0,0 +1,225 @@
+	work_damage_amount = 12
+	work_damage_type = WHITE_DAMAGE
+	ego_list = list(
+		/datum/ego_datum/armor/harmony
+		)
+	pixel_x = -8
+	base_pixel_x = -8
+	buckled_mobs = list()
+	buckle_lying = TRUE
+	var/cleanliness = "clean"
+	var/statChecked = 0
+	var/bonusRed = 0
+	var/grindRed = 4
+	var/minceRed = 8
+	var/playTiming = 5 SECONDS
+	var/playLength = 60 SECONDS
+	var/playStatus = 0
+	var/playRange = 20
+	var/datum/looping_sound/singing_grinding/grindNoise
+	var/datum/looping_sound/singing_music/musicNoise
+	var/list/musicalAddicts = list()
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/Life()
+	if(playStatus > 0) // If playstatus isn't 0, deal some damage in range.
+		for(var/mob/living/carbon/human/H in livinginrange(playRange, src))
+			if(faction_check_mob(H))
+				continue
+			H.apply_damage(rand(playStatus * 1, playStatus * 2), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+			if(H in musicalAddicts)
+				H.apply_damage(rand(playStatus * 1, playStatus * 2), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+				to_chat(H, "<span class='warning'>You can hear it again... it needs more...</span>")
+			else
+				to_chat(H, "<span class='warning'>That terrible grinding noise...</span>")
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/attempt_work(mob/living/carbon/human/user, work_type)
+	if(work_type == ABNORMALITY_WORK_INSTINCT)
+		if(datum_reference.qliphoth_meter > 0) // Sets bonus damage on instinct work.
+			bonusRed = grindRed // Should be weaker, for less lethal grinding.
+			manual_emote("makes an odd grinding noise...")
+			playsound(src, 'sound/abnormalities/singingmachine/grind.ogg', 40, 0, 1)
+		else
+			bonusRed = minceRed // Should be stronger, for when working it is EXTRA dangerous.
+			manual_emote("makes a horrible grinding noise!") // Oh boy, it's mad.
+			playsound(src, 'sound/abnormalities/singingmachine/chew.ogg', 60, 0, 3)
+	else if(get_attribute_level(user, FORTITUDE_ATTRIBUTE) >= 80 || get_attribute_level(user, TEMPERANCE_ATTRIBUTE) < 60)
+		statChecked = 1 // I see you've failed one of the stat checks, but have also chosen not to feed yourself to the machine.
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/work_chance(mob/living/carbon/human/user, chance)
+	chance -= statChecked * 15 // Perish.
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/worktick(mob/living/carbon/human/user)
+	if(bonusRed) // If you have bonus red damage to apply...
+		user.apply_damage(bonusRed, RED_DAMAGE, null, user.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
+		if(bonusRed < 6 && playStatus == 0)	// Should only happen when the machine isn't dealing damage.
+			for(var/mob/living/carbon/human/H in livinginrange(30, src))
+				if(faction_check_mob(H))
+					continue
+				H.adjustSanityLoss(rand(1,2))
```

Comment:
This INCREASES sanity of people in a very big range. Is it the intended effect?

Reply:
Yes, the intent is that when Singing Machine is "happy" its music is moderately helpful. The sanity healing can be toned down.